Mathlib/Data/ZMod/Basic.lean

```diff
@@ -981,6 +987,20 @@ theorem val_eq_zero : ∀ {n : ℕ} (a : ZMod n), a.val = 0 ↔ a = 0
 theorem val_ne_zero {n : ℕ} (a : ZMod n) : a.val ≠ 0 ↔ a ≠ 0 :=
   (val_eq_zero a).not
 
+theorem val_pos_of_ne_zero {n : ℕ} {a : ZMod n} (h : a ≠ 0) : 0 < a.val := by
+  apply Nat.pos_of_ne_zero
+  intro h
+  rw [ZMod.val_eq_zero] at h
+  contradiction
+
+theorem val_eq_one : ∀ {n : ℕ} [n.AtLeastTwo] (a : ZMod n), a.val = 1 ↔ a = 1
+  | 0, ⟨hn⟩, _
+  | 1, ⟨hn⟩, _ => by simp at hn
+  | n + 2, _, _ => by
+    rw [Fin.ext_iff]
+    simp only [Fin.val_one]
+    exact Iff.rfl
```

Comment:
I'm surprised simp doesn't cover this. Are you forcing through a defeq issue?

Reply:
Yes, `Fin.ext_iff` doesnt get applied by `simp` because the variables are still in `ZMod (n + 2)` instead of `Fin (n + 2)`. Likewise, the `Iff.rfl` needs to be forced because it's `Fin.val` on one side and `ZMod.val` on the other.